### PR TITLE
Add an interface to WeightedExperiment

### PR DESCRIPTION
**Pull request by jschueller**
This avoids to carry pointers to WeightedExperiment in various places.
It fixes the usage of SobolIndicesAlgorithm with an experiment.

### CODE REVIEW

File: lib/etc/openturns.conf.in [hunk 1]
```diff
@@ -462,8 +462,8 @@
   <MaternModel-DefaultNu    value="1.5"   />
   <MaternModel-DefaultTheta value="10.0"  />
   
-  <!-- OT::WeightedExperiment parameters -->
-  <WeightedExperiment-DefaultSize value="100" />
+  <!-- OT::WeightedExperimentImplementation parameters -->
+  <WeightedExperimentImplementation-DefaultSize value="100" />
```
Comment (dbarbier) [inline lib/etc/openturns.conf.in]: IMHO we should remove `Implementation` substrings from all resource keys (not in this PR, this is a general comment).  It does not add any value, and is annoying when performing changes like this one.

**Comment by dbarbier** [comment]
+1

File: lib/etc/openturns.conf.in (hunk 1 above)
Comment (jschueller) [inline lib/etc/openturns.conf.in]: @dbarbier agreed, will do it before branching

**Comment by sofianehaddad** [comment]
+1

**Comment by regislebrun** [comment]
+1

**Pull request merged**